src/hooks/use-has-user-interacted.ts

```diff
@@ -0,0 +1,4 @@
+import { hasUserInteracted } from '../utils.js'
+
+/** Returns `true` if user has interacted with the dialog. */
+export const useHasUserInteracted = (): boolean => hasUserInteracted()
```

Comment:
This hook should include something that makes it update when user does interact with the CMP. The way it's now it will only refresh the value on re-render, but not actively trigger updates on its own.

I think simply using the context would be enough, since the context value updates based on the events emitted by the CMP:

```ts
useContext(UsercentricsContext)
```

Reply:
Similar usage can be found here:

https://github.com/s-group-dev/react-usercentrics/blob/0f823288294b91eed763f656eaa9e2b88027c18b/src/hooks/use-service-info.ts#L19-L22

Reply:
Under the hood, `useServiceDebug` uses the `ping` context value:

https://github.com/s-group-dev/react-usercentrics/blob/0f823288294b91eed763f656eaa9e2b88027c18b/src/hooks/use-service-debug.ts#L8

Might be necessary to wrap in useMemo:

```ts
return useMemo(hasUserInteracted, [ping]
```